core/common/src/test/java/alluxio/EmbeddedChannels.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import com.google.common.base.Throwables;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelOutboundBuffer;
+import io.netty.channel.embedded.EmbeddedChannel;
+
+import java.lang.reflect.Method;
+
+/**
+ * Special versions of {@link EmbeddedChannel}.
+ */
+public final class EmbeddedChannels {
+  private EmbeddedChannels() {}  // prevent instantiation
+
+  /**
+   * A customized {@link EmbeddedChannel} that supports empty constructor and supports thread safe
+   * read/write outbound messages from/to this channel.
+   */
+  public static final class EmbeddedChannelEmptyCtor extends EmbeddedChannel {
```

Comment:
I guess I was just wondering what exactly this version of `EmbeddedChannel` was doing, and maybe name it for what it does, and not necessarily what the constructor looks like.